server/app/services/geo/esri/EsriClient.java

```diff
@@ -97,126 +102,102 @@ public CompletionStage<AddressSuggestionGroup> getAddressSuggestions(Address add
+                        .setAddressSuggestions(suggestionBuilder.build())
+                        .setOriginalAddress(address)
+                        .build();
+
+                // Record the execution time of the esri lookup process.
+                timer.observeDuration();
+                return addressCandidates;
+              } catch (JsonProcessingException e) {
+                logger.error(
+                    "EsriClient.fetchAddressSuggestions could not parse JSON response. Called by"
+                        + " EsriClient.getAddressSuggestions. Address = {}",
+                    address);
+                ESRI_LOOKUP_COUNT.labels("No suggestions").inc();
+                return AddressSuggestionGroup.empty(address);
+              }
             });
   }
 
-  /** Maps the JSON values within the node to an address */
-  @VisibleForTesting
-  static Address mapAddressAttributesJson(JsonNode candidateJson, Address address) {
-    JsonNode attributes = candidateJson.get("attributes");
-
+  static Address mapAddressAttributesJson(Attributes attributes, Address address) {
     return Address.builder()
-        .setStreet(getNodeTextValueOrEmpty(attributes, "Address"))
+        .setStreet(attributes.address())
         // If there is no value in SubAddr return the value the user entered. It may not be
         // returned by the ESRI instance and we don't want to lose the value.
-        .setLine2(getNodeTextValueOrDefault(attributes, "SubAddr", address::getLine2))
-        .setCity(getNodeTextValueOrEmpty(attributes, "City"))
+        .setLine2(
+            attributes.subAddr().isPresent() ? attributes.subAddr().get() : address.getLine2())
+        .setCity(attributes.city())
         .setState(getStateAbbreviationFromAttributes(attributes, address))
-        .setZip(getNodeTextValueOrEmpty(attributes, "Postal"))
+        .setZip(attributes.postal())
         .build();
   }
 
   /**
-   * Get the state abbreviation from the attributes JSON node. On Public ESRI this is under the
-   * RegionAbbr field, but some custom implementations may have it under the Region field.
+   * Get the state abbreviation. If the value from ESRI is not the two character state abbreviation,
+   * fallback to using the state that was originally set by the user.
    */
-  private static String getStateAbbreviationFromAttributes(JsonNode attributes, Address address) {
-    // Start with getting the value from the RegionAbbr, this is the
-    // default used by ESRI for state abbreviations
-    String result = getNodeTextValueOrEmpty(attributes, "RegionAbbr");
-
-    // Assume if it's not two characters it is either empty or the
-    // full state name and pull from the Region value
-    if (result.length() != 2) {
-      result = getNodeTextValueOrEmpty(attributes, "Region");
-    }
+  private static String getStateAbbreviationFromAttributes(Attributes attributes, Address address) {
+    String result = attributes.stateAbbreviation();
```

Comment:
question (non-blocking): do we still need to handle the case below where `result.length() != 2` since that is now handled when calling `attributes.stateAbbreviation()`?

I also have a recommendation to rename this to "region" like you have it in `Attributes` unless it is always state. And drop the "abbreviated" part of this method name since it's not always abbreviated.

Reply:
Yes it's still needed, I didn't want to make the fallback logic into the attributes as it is unrelated. So it still needs to fallback to the original user supplied state value here. I switched the attributes method to return empty if the abbreviation isn't two characters and now just check if it's present here or not.

I'm keeping the naming as state instead of region, for now. I know there are some assumptions in the system around addresses that it's going to be a US address. I'd rather wait to get naming right when we have a use case to test with other addresses.